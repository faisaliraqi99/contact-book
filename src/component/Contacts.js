import React from 'react';
import { Link } from 'react-router-dom';

import './Contact.css';
import contactIcon from '../img/phone-book.svg';

const Contacts = (props) => {
        return (
            <ul className="contact-list">
                {props.contacts.map((contact) =>
                    <li id={contact.id} onClick={props.selectItem} className="contact-item" key={contact.id}>
                        <Link className="contact-item-link" to="/details">
                            <img className="contact-item__img" src={contactIcon} alt="phone"></img>
                            <div className="contact-item__info">
                                <h3 className="contact-item__title">{contact.name} {contact.lastname}</h3>
                                <p className="contact-item__desc">Number: {contact.number}</p>
                            </div>
                        </Link>
                    </li>
                )}
            </ul>
        );
}

export default Contacts;