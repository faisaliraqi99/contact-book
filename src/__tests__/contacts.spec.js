import React from 'react'
import Contacts from '../component/Contacts'
import { shallow } from 'enzyme'

const initialData = [
  {
    id: 1,
    name: 'Somename',
    lastname: 'Lastname',
    number: '550121212',
    countrycode: '+996',
    email: 'undefined@mail.ru',
    address: 'Someaddr'
  },
  {
    id: 2,
    name: 'Somename',
    lastname: 'Lastname',
    number: '550121212',
    countrycode: '+996',
    email: 'undefined@mail.ru',
    address: 'Someaddr'
  }
]

describe('Contacts test', () => {

  const ContactsShallow = shallow(<Contacts contacts={initialData} />);

  it('Contacts is render from state snapshot', () => {
    expect(ContactsShallow).toMatchSnapshot();
  });

  it('Contacts correct render sum of contacts', () => {
    expect(ContactsShallow.find('li')).toHaveLength(initialData.length);
  });

  it('Contacts correct render data of props', () => {
    expect(ContactsShallow
      .find('li').first().find('h3').text())
      .toEqual(`${initialData[0].name} ${initialData[0].lastname}`);
  });

});