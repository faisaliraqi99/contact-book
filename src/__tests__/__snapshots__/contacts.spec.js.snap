// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts test Contacts is render from state snapshot 1`] = `
<ul
  className="contact-list"
>
  <li
    className="contact-item"
    id={1}
    key="1"
  >
    <Link
      className="contact-item-link"
      to="/details"
    >
      <img
        alt="phone"
        className="contact-item__img"
        src=""
      />
      <div
        className="contact-item__info"
      >
        <h3
          className="contact-item__title"
        >
          Somename
           
          Lastname
        </h3>
        <p
          className="contact-item__desc"
        >
          Number: 
          550121212
        </p>
      </div>
    </Link>
  </li>
  <li
    className="contact-item"
    id={2}
    key="2"
  >
    <Link
      className="contact-item-link"
      to="/details"
    >
      <img
        alt="phone"
        className="contact-item__img"
        src=""
      />
      <div
        className="contact-item__info"
      >
        <h3
          className="contact-item__title"
        >
          Somename
           
          Lastname
        </h3>
        <p
          className="contact-item__desc"
        >
          Number: 
          550121212
        </p>
      </div>
    </Link>
  </li>
</ul>
`;
